coordinator/src/services/translation/request/agent-unattended.ts

```diff
@@ -3,58 +3,72 @@ import {getIdentifierValueForSystem} from "../common"
 import {convertAddress, convertTelecom} from "./demographics"
 import pino from "pino"
 import {odsClient} from "../../communication/ods-client"
+import Hapi from "@hapi/hapi"
+import {
+  getRoleCode,
+  getSdsRoleProfileId,
+  getSdsUserUniqueId,
+  getUserName
+} from "../../../utils/headers"
 
 export async function createAuthorForUnattendedAccess(
   organizationCode: string,
-  logger: pino.Logger
+  headers: Hapi.Util.Dictionary<string>,
+  logger: pino.Logger,
+  telecom?: string
 ): Promise<hl7V3.Author> {
-  const agentPerson = await createAgentPersonForUnattendedAccess(organizationCode, logger)
+  const agentPerson = await createAgentPersonForUnattendedAccess(organizationCode, headers, logger, telecom)
   const author = new hl7V3.Author()
   author.AgentPerson = agentPerson
   return author
 }
 
 export async function createAgentPersonForUnattendedAccess(
   organizationCode: string,
-  logger: pino.Logger
+  headers: Hapi.Util.Dictionary<string>,
+  logger: pino.Logger,
+  fhirTelecom?: string
 ): Promise<hl7V3.AgentPerson> {
+  const sdsRoleProfileId = getSdsRoleProfileId(headers)
+  const sdsJobRoleCode = getRoleCode(headers)
+  const sdsUserUniqueId = getSdsUserUniqueId(headers)
+  const name = getUserName(headers)
+
+  const organization = await odsClient.lookupOrganization(organizationCode, logger)
+  if (!organization) {
+    throw new errors.InvalidValueError(
+      `No organisation details found for code ${organizationCode}`,
+      "Parameters.parameter"
```

Comment:
The FHIR path won't always be `Parameters.parameter` annoyingly. If FHIR path is optional maybe leave it out here?